src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -10,36 +12,53 @@ import "@polymer/paper-dialog/paper-dialog";
 import { PaperDialogElement } from "@polymer/paper-dialog/paper-dialog";
 import { HomeAssistant } from "../../../types";
 import { getCardConfig, updateCardConfig } from "../common/data";
+import { fireEvent } from "../../../common/dom/fire_event";
 
 import "./hui-yaml-editor";
 import "./hui-yaml-card-preview";
 // This is not a duplicate import, one is for types, one is for element.
 // tslint:disable-next-line
 import { HuiYAMLCardPreview } from "./hui-yaml-card-preview";
+import { LovelaceCardEditor, LovelaceConfig } from "../types";
+import { YamlChangedEvent, ConfigValue } from "./types";
+
+const CUSTOM_TYPE_PREFIX = "custom:";
 
 export class HuiDialogEditCard extends LitElement {
   protected hass?: HomeAssistant;
   private _cardId?: string;
-  private _cardConfig?: string;
+  private _currentConfigYaml?: string;
+  private _originalConfigYaml?: string;
+  private _configElement?: LovelaceCardEditor | null;
   private _reloadLovelace?: () => void;
+  private _editorToggle?: boolean;
+  private _configValue?: ConfigValue;
 
   static get properties(): PropertyDeclarations {
     return {
       hass: {},
       cardId: {
         type: Number,
       },
-      _cardConfig: {},
+      _currentConfigYaml: {},
       _dialogClosedCallback: {},
+      _configElement: {},
+      _editorToggle: {},
     };
   }
 
   public async showDialog({ hass, cardId, reloadLovelace }) {
     this.hass = hass;
     this._cardId = cardId;
     this._reloadLovelace = reloadLovelace;
-    this._cardConfig = "";
-    this._loadConfig();
+    this._currentConfigYaml = "";
+    this._editorToggle = true;
+    this._configElement = undefined;
+    this._loadConfig().then(() => this._loadConfigElement());
+    this._originalConfigYaml = this._currentConfigYaml;
+    if (!this._configElement) {
```

Comment:
config element is loaded in parallel and only after the config has bene fetched. So this check will always pass.

Reply:
(because the value will be `undefined`